src/AutoMapper/QueryableExtensions/ExpressionBuilder.cs

```diff
@@ -229,39 +222,51 @@ private static int GetDepth(ExpressionRequest request, TypePairCount typePairCou
             return visitCount;
         }
 
-        private LambdaExpression DestinationConstructorExpression(TypeMap typeMap, Expression instanceParameter, LetPropertyMaps letPropertyMaps)
+        private LambdaExpression DestinationConstructorExpression(ExpressionRequest request, Expression instanceParameter, TypePairCount typePairCount, TypeMap typeMap, LetPropertyMaps letPropertyMaps)
         {
             var ctorExpr = typeMap.CustomCtorExpression;
             if (ctorExpr != null)
             {
                 return ctorExpr;
             }
             var newExpression = typeMap.ConstructorMap?.CanResolve == true
-                ? typeMap.ConstructorMap.NewExpression(instanceParameter, letPropertyMaps)
+                ? NewExpression(typeMap.ConstructorMap, request, instanceParameter, typePairCount, letPropertyMaps)
                 : New(typeMap.DestinationTypeToUse);
-
             return Lambda(newExpression);
         }
 
-        private class NewFinderVisitor : ExpressionVisitor
-        {
-            public NewExpression NewExpression { get; private set; }
 
-            protected override Expression VisitNew(NewExpression node)
+        public Expression NewExpression(ConstructorMap constructorMap, ExpressionRequest request, Expression instanceParameter, TypePairCount typePairCount, LetPropertyMaps letPropertyMaps)
+        {
+            var parameters = constructorMap.CtorParams.Select(map =>
             {
-                NewExpression = node;
-                return base.VisitNew(node);
-            }
+                var resolvedSource = ResolveExpression(map, request.SourceType, instanceParameter, letPropertyMaps);
```

Comment:
I guess what should happen here is the same as for property maps. But that seems like a big change.

Reply:
Yeah, let's see how this goes first.